React/Base/RCTModuleMethod.mm

```diff
@@ -516,7 +516,19 @@ - (id)invokeWithBridge:(RCTBridge *)bridge
   }
 
   // Invoke method
-  [_invocation invokeWithTarget:module];
+  if (NSThread.isMainThread) {
+    NSTimeInterval start = [NSDate timeIntervalSinceReferenceDate];
+    [_invocation invokeWithTarget:module];
+    NSTimeInterval duration = [NSDate timeIntervalSinceReferenceDate] - start;
+    if (duration > 0.020) {
+      RCTLogWarn(@"mainThreadWatchdog: invocation of [%@ %@] took %dms",
```

Comment:
Use self as the argument and it will use the existing `description` method below.